python/pyspark/rdd.py

```diff
@@ -1039,7 +1039,8 @@ def sample(
         >>> 6 <= rdd.sample(False, 0.1, 81).count() <= 14
         True
         """
-        assert fraction >= 0.0, "Negative fraction value: %s" % fraction
+        if not fraction >= 0:
+            raise ValueError("Fraction must be nonnegative.")
```

Comment:
Change `assert` to ValueError is right definately, I believe all `assert` in main code (espacailly, use when validation params) should be fix (test mode is okay)

[1] https://mail.python.org/pipermail/python-list/2013-November/810940.html

Reply:
`AssertionError` is replaced with a `ValueError`, but I think it maybe trivial to add it to migration-doc